src/platform_impl/macos/view.rs

```diff
@@ -414,16 +437,44 @@ extern "C" fn selected_range(_this: &Object, _sel: Sel) -> NSRange {
     util::EMPTY_RANGE
 }
 
+fn id_to_string(string: id) -> String {
+    unsafe {
+        let has_attr = msg_send![string, isKindOfClass: class!(NSAttributedString)];
+        let characters = if has_attr {
```

Comment:
Here the compiler infers that `has_attr` is a `bool` but that's wrong. [`isKindOfClass`](https://developer.apple.com/documentation/objectivec/1418956-nsobject/1418511-iskindofclass/) returns a `BOOL` which is not the same as a `bool`. Important: to check if a `BOOL` is true, you must compare it with 0 or compare it with `NO`.

```suggestion
        let has_attr: BOOL = msg_send![string, isKindOfClass: class!(NSAttributedString)];
        let characters = if has_attr != NO {
```

Reply:
Exactly.